spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ldap/embedded/EmbeddedLdapAutoConfiguration.java

```diff
@@ -107,6 +109,22 @@ public InMemoryDirectoryServer directoryServer() throws LDAPException {
 					this.embeddedProperties.getCredential().getUsername(),
 					this.embeddedProperties.getCredential().getPassword());
 		}
+
+		if (!this.embeddedProperties.getValidation().isEnabled()) {
+			config.setSchema(null);
+		}
+		else if (this.embeddedProperties.getValidation().getSchema() != null) {
+			Resource schemaLocation = this.embeddedProperties.getValidation().getSchema();
+			try {
+				config.setSchema(Schema.mergeSchemas(Schema.getDefaultStandardSchema(),
+						Schema.getSchema(schemaLocation.getFile())));
```

Comment:
I took a closer look at the `Schema` class and I think there's a simpler solution, delegate the whole file loading to that class... It does mean reverting the property to a `String`. I'll push a sample, let me know what you think!